include/Dungeon/Enemies/Bat.h

```diff
@@ -18,22 +17,24 @@ class Bat : public Dungeon::Enemy {
 
     void Update() override;
 
+    void UpdateFace();
+
 private:
     void        RandomMove();
     void        MoveBat();
     std::size_t m_Tick = 2;
 
+private:
     const glm::vec2 m_FrameSize = {24, 24};
     std::size_t     m_State = 0;
 
     bool                     m_NeedToMove = false;
-    std::vector<std::size_t> m_RandomPool = {0, 1, 2, 3};
+    std::vector<std::size_t> m_RandomPool = {};
+    std::vector<glm::vec2>   m_Movement = {{0, 1}, {1, 0}, {0, -1}, {-1, 0}};
```

Comment:
Please write a conversion function from direction to Game Position.
like this.
```suggestion
    std::map<Enemy::Direction, glm::ivec2> {
    {Enemy::Direction::UP, {0, -1}},
    {Enemy::Direction::RIGHT, {1, 0}},
    {Enemy::Direction::DOWN, {0, 1}},
    {Enemy::Direction::LEFT, {-1, 0}}
    }
```